website/source/docs/auth/kerberos.html.md

```diff
@@ -0,0 +1,224 @@
+
+```text
+$ vault auth enable \
+    -passthrough-request-headers=Authorization \
+    -allowed-response-headers=www-authenticate \
+    kerberos
+```
+   
+* Create a `keytab` for the Kerberos plugin:
+
+```text
+$ ktutil
+ktutil:  addent -password -p your_service_account@REALM.COM -e aes256-cts -k 1
+Password for your_service_account@REALM.COM:
+ktutil:  list -e
+slot KVNO Principal
+---- ---- ---------------------------------------------------------------------
+  1    1            your_service_account@REALM.COM (aes256-cts-hmac-sha1-96)
+ktutil:  wkt vault.keytab
+```
+   
+   The KVNO (`-k 1`) should match the KVNO of the service account. An error will show in the Vault logs if this is incorrect.
+   
+   Different encryption types can also be added to the `keytab`, for example `-e rc4-hmac` with additional `addent` commands.
+   
+   Then base64 encode it:
+   
+```text
+$ base64 vault.keytab > vault.keytab.base64
+```
+
+* Configure the Kerberos auth method with the `keytab` and 
+entry name that will be used to verify inbound login 
+requests:
+
+```text
+$ vault write auth/kerberos/config \
+    keytab=@vault_svc.keytab.base64 \
+    service_account="vault_svc"
+```
+   
+* Configure the Kerberos auth method to communicate with
+LDAP using the service account configured above. This is 
+a sample LDAP configuration. Yours will vary. Ensure you've
+first tested your configuration from the Vault server using 
+a tool like `ldapsearch`.
+
+```text
+$ vault write auth/kerberos/config/ldap \
+    binddn=vault_svc@MATRIX.LAN \
+    bindpass=$VAULT_SVC_PASSWORD \
+    groupattr=sAMAccountName \
+    groupdn="DC=MATRIX,DC=LAN" \
+    groupfilter="(&(objectClass=group)(member:1.2.840.113556.1.4.1941:={{.UserDN}}))" \
+    userdn="CN=Users,DC=MATRIX,DC=LAN" \
+    userattr=sAMAccountName \
+    upndomain=MATRIX.LAN \
+    url=ldaps://somewhere.foo
+```
+   
+The LDAP above relies upon the same code as the LDAP auth method.
+See [its documentation](https://www.vaultproject.io/docs/auth/ldap.html) 
+for further discussion of available parameters. 
+   
+* Configure the Vault policies that should be granted to those 
+who successfully authenticate based on their LDAP group membership.
+Since this is identical to the LDAP auth method, see 
+[Group Membership Resolution](https://www.vaultproject.io/docs/auth/ldap.html#group-membership-resolution)
+and [LDAP Group -> Policy Mapping](https://www.vaultproject.io/docs/auth/ldap.html#ldap-group-gt-policy-mapping)
+for further discussion.
+
+```text
+$ vault write auth/kerberos/groups/engineering-team \
+    policies=engineers
+```
+
+The above group grants the "engineers" policy to those who authenticate
+via Kerberos and are found to be members of the "engineering-team" LDAP
+group.
+
+## Authentication
+
+From a client machine with a valid `krb5.conf` and `keytab`, perform a command
+like the following:
+
+```text
+$ vault login -method=kerberos \
+    username=grace \
+    service=HTTP/my-service \
```

Comment:
I forget, does the service principal need to match a configuration value that was written to LDAP earlier? Or is that outside the scope of this documentation? (For instance, in integration_env.sh it's set up https://github.com/hashicorp/vault-plugin-auth-kerberos/blob/e7680d2ebcaa2ce167cafca78d8a7bd49ae88cfd/scripts/integration_env.sh#L121)

Reply:
Ah, good point! It does need to match one that was set up earlier. In our dev/test environment, for the value [here](https://github.com/hashicorp/vault-plugin-auth-kerberos/blob/master/cmd/login-kerb/main.go#L44) we are indeed matching upon the one we've already set up [here](https://github.com/hashicorp/vault-plugin-auth-kerberos/blob/master/scripts/integration_env.sh#L125). 

Yes, I think you're right that we should point out these values need to match service accounts that have already been set up in LDAP. I'll avoid _giving folks the commands_ to do it because the commands will vary across their environments. If we give the Samba example, then people will likely ask us to give more examples for more environments, and that would be daunting.

Reply:
Added in https://github.com/hashicorp/vault/pull/7993/commits/ba423788b7fa77c7615b6772e67e673d42dbee3b